Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -257,31 +239,20 @@ class DatabaseContainer: NSPersistentContainer {
         }
 
         writableContext.perform { [weak self] in
-            self?.sendNotificationForAllContexts(name: Self.WillRemoveAllDataNotification)
-
-            // Because this is a critical part of Stream, we don't want to make such changes unless it is in a controlled environment
-            if StreamRuntimeCheck._isBackgroundMappingEnabled {
-                // Before we remove the persistent store, we need to reset the state of the `NSManagedObjectContext`s to
-                // avoid a context from trying to access data from a previous store
-                self?.allContext.forEach { context in
-                    context.performAndWait { context.reset() }
+            let entityNames = self?.managedObjectModel.entities.compactMap(\.name)
+            var deleteError: Error?
+            entityNames?.forEach { [weak self] entityName in
+                let deleteFetch = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
+                let deleteRequest = NSBatchDeleteRequest(fetchRequest: deleteFetch)
+                do {
+                    try self?.writableContext.execute(deleteRequest)
+                    try self?.writableContext.save()
```

Comment:
Does it make sense to only save once all entities have been completed? In that way we can at least have a state where we can clearly interact when we try to login.

Reply:
Hmm, what if things fail, or the app is killed by the user? We won't have anything deleted. At least in this way we delete step by step (we could also put the higher prio ones first?)

Reply:
If things fail (I'm guessing that this if the app gets killed before it completes) that's ok because then we come back, on login(or even on startup) we can erase everything.

Reply:
The performAndWait will probably make sure we are safer against not finishing the operation. But we might be overthinking here, we clear the data again whenever we login with a different user, so it is fine.